### DISCUSSION TITLE
CSPDarknet53 as a Backbone

### DISCUSSION THREAD

**Discussion by ErikValle** [ultralytics/yolov5#6395] [Q&A] [answered]
The YOLOv4 [paper](https://arxiv.org/pdf/2004.10934v1.pdf) suggests using a backbone based on CSPResNext50 or CSPDarknet53. The first one has a higher accuracy when trained with different features, while the other shows higher accuracy in object detection. Since the yolov5x YAML file follows a kind of CSPDarknet53, I expected to see something following this [architecture](https://production-media.paperswithcode.com/methods/Screen_Shot_2020-06-24_at_12.53.56_PM_QQoF5AO.png) but with CSP. Nevertheless, it has the elements below: 

### YOLOv5 backbone
backbone:
  [[-1, 1, Focus, [64, 3]],  # 0-P1/2
   [-1, 1, Conv, [128, 3, 2]],  # 1-P2/4
   [-1, 3, C3, [128]],
   [-1, 1, Conv, [256, 3, 2]],  # 3-P3/8
   [-1, 9, C3, [256]],
   [-1, 1, Conv, [512, 3, 2]],  # 5-P4/16
   [-1, 9, C3, [512]],
   [-1, 1, Conv, [1024, 3, 2]],  # 7-P5/32
   [-1, 1, SPP, [1024, [5, 9, 13]]],
   [-1, 3, C3, [1024, False]],  # 9
  ]

## Questions:
1. Shouldn't the backbone include a 32-depth Conv block as listed below? I tried to adjust the architecture as in [CSPDarknet53](https://www.researchgate.net/publication/351731859/figure/fig2/AS:1025659312816128@1621547644217/The-structure-of-CSPDarknet53-a-and-CSPDarknet53-tiny-b.ppm) but it may be wrong
### YOLOv5 backbone (my suggestion)
backbone:
   [[-1, 1, Conv, [32, 3]],  # 0-P1/2 
   [-1, 1, Conv, [64, 3, 2]],  # 1-P2/4 
   [-1, 3, C3, [128]],
   [-1, 1, Conv, [128, 3, 2]],  # 3-P3/8
   [-1, 9, C3, [256]],
   [-1, 1, Conv, [256, 3, 2]],  # 5-P4/16
   [-1, 9, C3, [512]],
   [-1, 1, Conv, [512, 3, 2]],  # 7-P5/32
   [-1, 9, C3, [1024]],
   [-1, 1, Conv, [1024, 3, 2]],  # 9-P6/64
   [-1, 9, C3, [2048]],
   [-1, 1, SPP, [2048, [5, 9, 13]]],
   [-1, 3, C3, [1024, False]],  # 11
  ]
2. Why did you use Focus as the first operation instead of Conv? I was just curious, you guys have already discussed it here: https://github.com/ultralytics/yolov5/issues/4825
Thank you

**Comment by glenn-jocher** [comment, marked as answer]
@ErikValle YOLOv5 backbone and head have evolved over releases and include C3 modules which are reduced CSP modules. 

Focus conversation is fully covered in https://github.com/ultralytics/yolov5/issues/4825

**Reply by ErikValle** [reply]
Thank you!
